php/class-customize-snapshot-manager.php

```diff
@@ -410,6 +286,8 @@ public function create_initial_changeset_revision( $post_id ) {
 	/**
 	 * Prepare snapshot post content for publishing.
 	 *
+	 * @todo Move to compat?
```

Comment:
I don't think so. The idea for the `publish_error` is that if you have a changeset saved as a draft, but then you add a new validation constraint or remove a setting from being registered, that if you then try to publish the changeset from the admin, that the changeset will get blocked from publishing and an error message will then be displayed when editing the post.

Reply:
Humm. I can see that it will be somewhat challenging to restore this `publish_error` functionality in 4.9 because `save_settings_with_publish_snapshot` in the plugin is now `\WP_Customize_Manager::_publish_changeset_values` in core. We'd have to somehow suspend the `_wp_customize_publish_changeset` function from running during the `transition_post_status` action when it detects that there is an invalid setting.

For example:

```php
add_action( 'transition_post_status', function( $new_status, $old_status, $changeset_post ) {
   // @todo Check if there are any invalid settings in $changeset_post...
   $has_invalid_settings = true; // ...

   // Prevent publishing of settings.
   if ( $has_invalid_settings ) {
        remove_action( 'transition_post_status', '_wp_customize_publish_changeset' );
        add_action( 'transition_post_status', function( $n, $o, $post ) {
            if ( $post->ID === $changeset_post->ID ) {
                add_action( 'transition_post_status', '_wp_customize_publish_changeset' );
            }
        }, 100, 3 ); // later
    }
}, 9, 3 ); // Before _wp_customize_publish_changeset() at priority 10.

````

Reply:
If its not very important right now, can I do this in a separate PR, I spent some time but could't get my head around this very well, I need to look back and forth to better understand, and since `show_publish_error_admin_notice`  and `display_post_states` were kept compat `php/class-post-type-back-compat.php`, so I assume its not being used for 4.7 anyways, though I have restored them.

Reply:
Yes, we can restore that functionality later. Normally it should not be used. It's more of an edge case.